src/model/openthaigpt_pretraining_model/models/nanoGPT/model.py

```diff
@@ -28,8 +59,490 @@ def _attn_wrapper(self, query, key, value, attention_mask=None, head_mask=None):
     return attn_out, None
 
 
+class EditGPT2Attention(OriginalGPT2Attention):
+    def __init__(self, config, is_cross_attention=False, layer_idx=None):
+        super().__init__(config)
+
+        self.rotary_pct = ROTARY_PCT
+        self.rotary_emb_base = ROTARY_EMB_BASE
+        self.use_rotary = config.use_rotary
+        self.rotary_ndims = int(self.head_dim * self.rotary_pct)
+        if self.use_rotary:
+            self.rotary_emb = RotaryEmbedding(
+                self.rotary_ndims,
+                config.max_position_embeddings,
+                base=self.rotary_emb_base,
+            )
+
+    def forward(
```

Comment:
You should add the comment where are you edited from original source code. You can see an example in [Electra repo](https://github.com/OpenThaiGPT/Pretraining-ELECTRA/blob/36acaf52364c4159adfdb2a5e975bd2600f92315/src/electra_pretraining/electra/flash_attn_model.py#L189).

Reply:
[DONE] but if you want to make sure that where I edit and add more.... you can search "self.use_rotary" that is the place I'm woking on.